spoon-client/src/main/java/com/squareup/spoon/Spoon.java

```diff
@@ -229,19 +229,34 @@ private static File filesDirectory(Context context, String directoryType, String
   static StackTraceElement findTestClassTraceElement(StackTraceElement[] trace) {
     for (int i = trace.length - 1; i >= 0; i--) {
       StackTraceElement element = trace[i];
-
+      int apiVersion = Build.VERSION.SDK_INT;
       if (TEST_CASE_CLASS_JUNIT_3.equals(element.getClassName()) //
           && TEST_CASE_METHOD_JUNIT_3.equals(element.getMethodName())) {
-        return trace[i - 3];
+          //Stacktrace length changed in M
```

Comment:
lots of dup code.  can you pull this into a helper method called `extractStackElement` or somesuch?